File: 04 - ORIENTAÇÃO A OBJETOS/Java/Exercicio 3/TestePessoa.java

public class TestePessoa {
    public static void main(String[] args) {
        Pessoa pe1 = new Pessoa("João", 21, 75.4);
        Pessoa pe2 = new Pessoa("Carlos");
        System.out.println(pe1.dizerInfo());
        System.out.println(pe2.dizerInfoNome());
        
        Carro car = new Carro();
        System.out.println("\nPiloto do carro 1: "+car.piloto.nome);
        Carro car2 = new Carro(pe1);
        System.out.println("Piloto do carro 2: "+car2.piloto.dizerInfo());
    }
}

File: 04 - ORIENTAÇÃO A OBJETOS/nodeJS - JavaScript_/Exercicio 3/Com Pacotes/TestePessoa.js
let Pessoa = require('./Pessoa')
let Carro = require('./Carro')

let pe1 = new Pessoa('João', 21, 75.3)
let pe2 = new Pessoa('Carlos')
console.log(pe1.dizerInfo())
console.log(pe2.dizerNome())

//Os parametros do construtor devem seguir conforme a ordem de passagem.
let car = new Carro('Marcos')
console.log('\nPiloto do carrro 1: '+car.piloto)
let car2 = new Carro(pe1)
console.log('Piloto do carrro 2: '+car2.piloto.dizerInfo())
